lib/assertions.js

```diff
@@ -757,16 +757,70 @@ module.exports = function (expect) {
         '<object> to be (a map|a hash|an object) whose (keys|properties) satisfy <any+>',
         '<object> to be (a map|a hash|an object) whose (keys|properties) satisfy <assertion>'
     ], function (expect, subject) {
-        var extraArgs = Array.prototype.slice.call(arguments, 2);
         expect.errorMode = 'nested';
         expect(subject, 'to be an object');
         expect(subject, 'not to equal', {});
         expect.errorMode = 'default';
 
         var keys = expect.subjectType.getKeys(subject);
+        var extraArgs = Array.prototype.slice.call(arguments, 2);
         return expect.apply(expect, [keys, 'to have items satisfying'].concat(extraArgs));
     });
 
+    expect.addAssertion([
+        '<object> to have a value [exhaustively] satisfying <any+>',
```

Comment:
Should be `<any>` as subsequent args won't be used, except in the `<assertion>` case.

I realize this was copy/pasted from the neighbor assertions -- I fixed those here: https://github.com/unexpectedjs/unexpected/pull/348